packages/rocketchat-api/server/v1/channels.js

```diff
@@ -747,6 +747,28 @@ RocketChat.API.v1.addRoute('channels.setCustomFields', { authRequired: true }, {
 	}
 });
 
+RocketChat.API.v1.addRoute('channels.setDefault', { authRequired: true }, {
+	post() {
+		if (typeof this.bodyParams.default === 'undefined') {
+			return RocketChat.API.v1.failure('The bodyParam "default" is required');
+		}
+
+		const findResult = findChannelByIdOrName({ params: this.requestParams() });
+
+		if (findResult.default === this.bodyParams.default) {
+			return RocketChat.API.v1.failure('The channel default setting is the same as what it would be changed to.');
```

Comment:
Missing localization key here

```javascript
return RocketChat.API.v1.failure('The channel default setting is the same as what it would be changed to.', 'error-channels-setdefault-is-same');
```